lxd/certificates.go

```diff
@@ -377,6 +380,20 @@ func clusterMemberJoinTokenValid(d *Daemon, r *http.Request, projectName string,
 			return nil, fmt.Errorf("Failed to cancel operation %q: %w", foundOp.ID, err)
 		}
 
+		var expiry time.Time
+
+		// Depending on whether it's a local operation or not, expiry will either be a time.Time or a string.
+		if d.serverName == foundOp.Location {
+			expiry, _ = foundOp.Metadata["expiresAt"].(time.Time)
+		} else {
+			expiry, _ = time.Parse(time.RFC3339Nano, foundOp.Metadata["expiresAt"].(string))
+		}
+
+		// Check if token has expired.
+		if time.Now().After(expiry) {
```

Comment:
Is it right that if no expiry is found the token is assumed to have expired?

Reply:
That would be the case. I'll add a check for `foundOp.Metadata["expiresAt"]`.